### PR TITLE
Closes #219 | Create dataloader for scb-mt-en-th-2020

### LINKED ISSUE #219: Create dataset loader for scb-mt-en-th-2020
Dataloader name: `scb-mt-en-th/scb-mt-en-th.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?scb-mt-en-th

|     Dataset| scb-mt-en-th |
|-------------|---|
| Description | A Large English-Thai Parallel Corpus The primary objective of our work is to build a large-scale English-Thai dataset for machine translation. We construct an English-Thai machine translation dataset with over 1 million segment pairs, curated from various sources, namely news, Wikipedia articles, SMS messages, task-based dialogs, web-crawled data and government documents. Methodology for gathering data, building parallel texts and removing noisy sentence pairs are presented in a reproducible manner. We train machine translation models based on this dataset. Our models' performance are comparable to that of Google Translation API (as of May 2020) for Thai-English and outperform Google when the Open Parallel Corpus (OPUS) is included in the training data for both Thai-English and English-Thai translation. The dataset, pre-trained models, and source code to reproduce our work are available for public use. |
| Subsets | - |
| Languages | tha, eng |
| Tasks | Machine Translation |
| License | Creative Commons Attribution Share Alike 4.0 (cc-by-sa-4.0) |
| Homepage | https://github.com/vistec-AI/thai2nmt |
| HF URL | https://huggingface.co/datasets/scb_mt_enth_2020 |
| Paper URL | https://link.springer.com/article/10.1007/s10579-021-09536-6 |

### PR DESCRIPTION
**Pull request by jen-santoso**
Please name your PR after the issue it closes. You can use the following line: "Closes #219 " where you replace the ISSUE-NUMBER with the one corresponding to your dataset.

### Checkbox
- [ ] Confirm that this PR is linked to the dataset issue.
- [ ] Create the dataloader script `seacrowd/sea_datasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [ ] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [ ] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [ ] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [ ] Confirm dataloader script works with `datasets.load_dataset` function.
- [ ] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py`.
- [ ] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

**Comment by jen-santoso** [comment]
Hello reviewers,

In addition to the PR, I would like to report that the DataCatalogue link in #219 does not work.
Instead, I found this link works for this dataset: https://seacrowd.github.io/seacrowd-catalogue/card.html?scb-mt-en-th-2020.

Therefore, one clarification: Should the dataset name be scb-mt-en-th or scb-mt-en-th-2020?
I think the dataset name reported in the title, card, and dataloader name should be unified.
Please let me know which dataset name is the correct one.

**Comment by holylovenia** [comment]
> Hello reviewers,
> 
> In addition to the PR, I want to let you know (as reported in the comment section of issue #219) that the DataCatalogue link cannot be opened from here. Instead, I found this link works for this dataset: https://seacrowd.github.io/seacrowd-catalogue/card.html?scb-mt-en-th-2020.
> 
> Therefore, one clarification: Should the dataset name be scb-mt-en-th or scb-mt-en-th-2020? (UPDATE: it will be implemented with the `snakecase` dataloader name) I think the dataset name reported in the title, card, and dataloader name should be unified. Please let me know which dataset name is the correct one.

Hi @jen-santoso, let's stick with `scb_mt_en_th`.
I'll run this by @SamuelCahyawijaya to see what triggers the incorrect URL being generated.

File: seacrowd/sea_datasets/scb_mt_en_th/scb_mt_en_th.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+            schema="seacrowd_t2t",
+            subset_id=f"{_DATASETNAME}",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = f"{_DATASETNAME}_tha_eng_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+        if self.config.schema == "source":
+            language_pair = [SEACROWD_TO_SOURCE_LANGCODE_DICT[lang] for lang in self.config.name.split("_")[1:3]]
```
Comment (holylovenia) [inline seacrowd/sea_datasets/scb_mt_en_th/scb_mt_en_th.py:112]: `1:3` --> `4:6`

File: seacrowd/sea_datasets/scb_mt_en_th/scb_mt_en_th.py [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+        return [
+            datasets.SplitGenerator(name=datasets.Split.TRAIN, gen_kwargs={"filepath": os.path.join(data_dir, "train.jsonl")}),
+            datasets.SplitGenerator(name=datasets.Split.VALIDATION, gen_kwargs={"filepath": os.path.join(data_dir, "valid.jsonl")}),
+            datasets.SplitGenerator(name=datasets.Split.TEST, gen_kwargs={"filepath": os.path.join(data_dir, "test.jsonl")}),
+        ]
+
+    def _generate_examples(self, filepath: Path) -> Tuple[int, Dict]:
+        with open(filepath, encoding="utf-8") as f:
+            if self.config.schema == "source":
+                language_pair = [SEACROWD_TO_SOURCE_LANGCODE_DICT[lang] for lang in self.config.name.split("_")[1:3]]
```
Comment (holylovenia) [inline seacrowd/sea_datasets/scb_mt_en_th/scb_mt_en_th.py:144]: `1:3` --> `4:6`

File: seacrowd/sea_datasets/scb_mt_en_th/scb_mt_en_th.py [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+                source, target = language_pair
+                for id_, row in enumerate(f):
+                    data = json.loads(row)
+                    yield id_, {
+                        "translation": {source: data[source], target: data[target]},
+                        "subdataset": data["subdataset"],
+                    }
+
+            elif self.config.schema == "seacrowd_t2t":
+                source, target = self.config.name.split("_")[1:3]
```
Comment (holylovenia) [inline seacrowd/sea_datasets/scb_mt_en_th/scb_mt_en_th.py:154]: `1:3` --> `4:6`

**Review by holylovenia — CHANGES_REQUESTED** [review]
Hi @jen-santoso, thanks for your dataloader!! Could you please add `_LANGUAGES = ["tha", "eng"]`? Also, I've got a few suggestions above.

**Review by holylovenia — APPROVED** [review]
Thanks for the changes, @jen-santoso! Good work as always~ Let's wait for @jamesjaya's review.

**Review by jamesjaya — APPROVED** [review]
tested, lgtm

**Pull request merged**